pkg/service/model_build_load_balancer.go

```diff
@@ -0,0 +1,157 @@
+package service
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/pkg/errors"
+	"regexp"
+	"sigs.k8s.io/aws-load-balancer-controller/pkg/annotations"
+	elbv2model "sigs.k8s.io/aws-load-balancer-controller/pkg/model/elbv2"
+	"strconv"
+)
+
+const (
+	lbAttrsAccessLogsS3Enabled           = "access_logs.s3.enabled"
+	lbAttrsAccessLogsS3Bucket            = "access_logs.s3.bucket"
+	lbAttrsAccessLogsS3Prefix            = "access_logs.s3.prefix"
+	lbAttrsLoadBalancingCrossZoneEnabled = "load_balancing.cross_zone.enabled"
+
+	resourceIDLoadBalancer = "LoadBalancer"
+)
+
+func (t *defaultModelBuildTask) buildLoadBalancer(ctx context.Context, scheme elbv2model.LoadBalancerScheme) error {
+	spec, err := t.buildLoadBalancerSpec(ctx, scheme)
+	if err != nil {
+		return err
+	}
+	t.loadBalancer = elbv2model.NewLoadBalancer(t.stack, resourceIDLoadBalancer, spec)
+	return nil
+}
+
+func (t *defaultModelBuildTask) buildLoadBalancerSpec(ctx context.Context, scheme elbv2model.LoadBalancerScheme) (elbv2model.LoadBalancerSpec, error) {
+	ipAddressType := elbv2model.IPAddressTypeIPV4
+	lbAttributes, err := t.buildLoadBalancerAttributes(ctx)
+	if err != nil {
+		return elbv2model.LoadBalancerSpec{}, err
+	}
+	tags, err := t.buildLoadBalancerTags(ctx)
+	if err != nil {
+		return elbv2model.LoadBalancerSpec{}, err
+	}
+	subnetMappings, err := t.buildSubnetMappings(ctx, t.ec2Subnets)
+	if err != nil {
+		return elbv2model.LoadBalancerSpec{}, err
+	}
+	name := t.buildLoadBalancerName(ctx, scheme)
+	spec := elbv2model.LoadBalancerSpec{
+		Name:                   name,
+		Type:                   elbv2model.LoadBalancerTypeNetwork,
+		Scheme:                 &scheme,
+		IPAddressType:          &ipAddressType,
+		SubnetMappings:         subnetMappings,
+		LoadBalancerAttributes: lbAttributes,
+		Tags:                   tags,
+	}
+	return spec, nil
+}
+
+func (t *defaultModelBuildTask) buildLoadBalancerScheme(_ context.Context) (elbv2model.LoadBalancerScheme, error) {
+	scheme := elbv2model.LoadBalancerSchemeInternetFacing
+	internal := false
+	if _, err := t.annotationParser.ParseBoolAnnotation(annotations.SvcLBSuffixInternal, &internal, t.service.Annotations); err != nil {
+		return "", err
+	}
+	if internal {
```

Comment:
it would be more clear to me if it's like below. (the idea is to return asap)
```
if internal {
   return elbv2model.LoadBalancerSchemeInternal, nil
}
return elbv2model.LoadBalancerSchemeInternetFacing, nil
```